gammapy/catalog/gammacat.py

```diff
@@ -357,6 +358,10 @@ def spatial_model(self):
             raise NoDataAvailableError(f"No spatial model available: {self.name}")
         else:
             raise NotImplementedError(f"Unknown spatial model: {morph_type!r}")
+        model._position_error = CircleSkyRegion(
```

Comment:
It would be better if you change everywhere to set like this:
```
pos_err = CircleSkyRegion(...)
model.set_position_error(position_error)
```
For now, you can still store away in `model._position_error`.
But soon, we will change to store away in covar instead.
And if you put that setter helper method now, you don't have to update all callers later.

Generally, when you get or set a private property like `model._position_error` from outside the class, it's not a good and clean API and needs to be changes.

In this case, the best final API I can imagine is this error regions-based one, where we can get and set on the model object.